sympy/stats/rv_interface.py

```diff
@@ -208,6 +209,59 @@ def skewness(X, condition=None, **kwargs):
     """
     return smoment(X, 3, condition, **kwargs)
 
+def quantile(X, p):
+    r"""
+    Return the :math:`p^{th}` order quantile of a probability distribution.
+
+    Quantile is defined as the value at which the probability of the random
+    variable is less than or equal to the given probability.
+
+    ..math::
+        Q(p) = inf{x \in (-\infty, \infty) such that p < F(x)}
+
+    Examples
+    ========
+
+    >>> from sympy.stats import quantile, Die, Exponential
+    >>> from sympy import Symbol, pprint
+    >>> p = Symbol("p", positive=True)
+
+    >>> l = Symbol("lambda", positive=True)
+    >>> X = Exponential("x", l)
+    >>> pprint(quantile(X, p), use_unicode=False)
+               -log(1 - p)
+    [0, oo) n {------------}
+                  lambda
+
+    >>> D = Die("d", 6)
+    >>> pprint(quantile(D, p), use_unicode=False)
+    /1  for p <= 1/6
+    |
+    |2  for p <= 1/3
+    |
+    |3  for p <= 1/2
+    <
+    |4  for p <= 2/3
+    |
+    |5  for p <= 5/6
+    |
+    \6   for p < 1
+    """
+    _value_check(p > 0, "The order p must be positive.")
+    _value_check(p < 1, "The order p must be less than 1.")
+
+    if pspace(X).is_Continuous:
```

Comment:
Maybe here, I can write `if  pspace(X).is_Conitnuous or pspace(X).is_Discrete:`
It will cover discrete random variables as well in that case

Reply:
No, it will be wrong. I'll think of something else

Reply:
I have added support for discrete cases as well